### ISSUE TITLE
After installing Zygisk module, I get stuck on boot logo.

### ISSUE THREAD

**Issue by lss4** [Dr-TSNG/Hide-My-Applist#137]
**TL;DR: Hide My Applist REQUIRES LSPosed (Zygisk version) be installed. After installing Hide My Applist's Zygisk module (via command line), activate Hide My Applist inside LSPosed BEFORE reboot, or the system will BOOTLOOP!**

**PS: In case you can't easily access LSPosed menu, go to dialer, dial `*#*#5776733#*#*` to enter it.**

---

Device: Motorola Edge S Pro (pstar)
OS: DotOS 5.2 GSI (Android 11)
Hide-My-Applist Version: V2.3.2 Final
Magisk Version: Official Canary 24102 (25)

After installing Zygisk module, the system gets stuck on boot logo.
I cannot get logcat as the device could not even be seen by my PC in this state.

I'll see if I could boot to TWRP to get myself out of the mess, but is there any other way to inspect what had gone wrong?

**Comment by lss4** [comment]
Okay... was able to get `/data/adb/logcats` via TWRP.

Here's a portion of filtered logcat made by grepping the logcat file for "Magisk".

```
02-15 09:41:10.773  2162  2162 I Magisk  : * Launching zygiskd64
02-15 09:41:10.800  2155  2155 I [HMA Native]: Magisk extension start. Version 1.0 (35)
02-15 09:41:17.058  3536  3536 I Magisk  : * Launching zygiskd32
02-15 09:41:27.006  1235  1351 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-15 09:41:27.025  1235  1237 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-15 09:41:27.045  1235  3517 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-15 09:41:27.071  1235  3533 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-15 09:41:27.158  1235  3554 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-15 09:41:27.253  1235  3591 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-15 09:41:37.308  1235  3773 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-15 09:41:37.332  1235  1351 E Magisk  : read (4 != 0) failed with 2: No such file or directory
```

Note this:
```
02-15 09:41:37.332  1235  1351 E Magisk  : read (4 != 0) failed with 2: No such file or directory
```

When this Zygisk module is not installed, this error line would not appear the the system can boot properly.

Initially I thought Storage Isolation and Shamiko might be responsible, so I tried installing the module with these modules turned off and I still get this same issue, so their influence can be ruled out.

**Comment by keta1** [comment]
@lss4 I will confirm it soon

**Comment by tongdhW** [comment]
我也有同样的问题，重启后一直崩溃
Phone model: Pixel 2 XL

Android version: 8.1.0

ROM: AOSP on taimen

Magisk version: 24.1

List of Magisk modules:

Enhanced mode for HideMyApplist v1.0
LSPosed (Zygisk) v1.7.2

List of LSPosed modules:

Hide My Applist v2.3.2



02-24 11:34:57.394   682  1710 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-24 11:34:57.766   682  1783 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-24 11:34:58.461   682  1701 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-24 11:35:04.136   682   684 E Magisk  : read (4 != 0) failed with 2: No such file or directory
02-24 11:35:14.209   682  1673 E Magisk  : read (4 != 0) failed with 2: No such file or directory


02-24 11:35:45.746  2347  2347 E CHRE    : Failed to deliver timestamp message from host to CHRE: 59
02-24 11:35:45.753  2347  2347 D /vendor/bin/chre: vendor/qcom/proprietary/adsprpc/src/fastrpc_apps_user.c:1038: Error ffffffff: apps_dev_init failed. domain 2, errno Operation not permitted
02-24 11:35:45.753  2347  2347 D /vendor/bin/chre: vendor/qcom/proprietary/adsprpc/src/fastrpc_apps_user.c:1113: Error ffffffff: open dev -1 for domain 2 failed
02-24 11:35:45.761  2347  2347 D /vendor/bin/chre: vendor/qcom/proprietary/adsprpc/src/fastrpc_apps_user.c:1038: Error ffffffff: apps_dev_init failed. domain 2, errno Operation not permitted
02-24 11:35:45.761  2347  2347 D /vendor/bin/chre: vendor/qcom/proprietary/adsprpc/src/fastrpc_apps_user.c:1113: Error ffffffff: open dev -1 for domain 2 failed
02-24 11:35:45.761  2347  2347 D /vendor/bin/chre: vendor/qcom/proprietary/adsprpc/src/fastrpc_apps_user.c:571: Error 3b: remote handle invoke failed. domain 2, handle ffffffff, sc 0, pra 0x0
02-24 11:35:45.761  2347  2347 E CHRE    : Failed to start CHRE on SLPI: 59

**Comment by tongdhW** [comment]
同样的软件在另一台机器（Android 12定制机器）没问题

**Comment by tongdhW** [comment]
上面问题的log文件

[phone_pixel2xl.log](https://github.com/Dr-TSNG/Hide-My-Applist/files/8130008/phone_pixel2xl.log)
[magisk.log](https://github.com/Dr-TSNG/Hide-My-Applist/files/8130010/magisk.log)
[lsposed.log.tar.gz](https://github.com/Dr-TSNG/Hide-My-Applist/files/8130013/lsposed.log.tar.gz)

**Comment by lukemin** [comment]
安装完zygisk模块后，一定要在lsposed里面也把模块激活，要不然就会卡在boot

**Comment by lss4** [comment]
> 安装完zygisk模块后，一定要在lsposed里面也把模块激活，要不然就会卡在boot

Question:
I recall seeing that Zygisk and Riru+LSPosed are mutually exclusive. Does LSPosed have a Zygisk-compatible version?

**Comment by cracky5322** [comment]
> > 安装完zygisk模块后，一定要在lsposed里面也把模块激活，要不然就会卡在boot
> 
> Question: I recall seeing that Zygisk and Riru+LSPosed are mutually exclusive. Does LSPosed have a Zygisk-compatible version?

Please search LSPosed's Github and you will get the answer

**Comment by lss4** [comment]
Yes. It's working now. So... Hide My Applist apparently **REQUIRES** LSPosed (Zygisk version) be installed.

After installing Zygisk module using the command line, **activate Hide My Applist in LSPosed BEFORE rebooting or the system will BOOTLOOP!**

I think despite the project is not maintained anymore, this should be put as a heads-up warning in case any other user falls into the same pit as I did.

PS: In case you can't easily access LSPosed menu, go to dialer, dial `*#*#5776733#*#*` to enter it.

**Comment by lss4** [comment]
Will close the issue now. I've put the warning on the OP as well.
